sdk/src/types/block/protocol.rs

```diff
@@ -5,42 +5,65 @@ use alloc::string::String;
 use core::borrow::Borrow;
 
 use crypto::hashes::{blake2b::Blake2b256, Digest};
-use packable::{prefix::StringPrefix, Packable, PackableExt};
+use getset::{CopyGetters, Getters};
+use packable::{
+    prefix::{BoxedSlicePrefix, StringPrefix},
+    Packable, PackableExt,
+};
 
-use super::{address::Hrp, slot::SlotIndex};
+use super::{
+    address::Hrp,
+    slot::{EpochIndex, SlotIndex},
+};
 use crate::types::block::{helper::network_name_to_id, output::RentStructure, ConvertTo, Error, PROTOCOL_VERSION};
 
 /// Defines the parameters of the protocol.
-#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Packable)]
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Packable, Getters, CopyGetters)]
 #[packable(unpack_error = Error)]
 #[cfg_attr(
     feature = "serde",
     derive(serde::Serialize, serde::Deserialize),
     serde(rename_all = "camelCase")
 )]
+#[getset(get_copy = "pub")]
 pub struct ProtocolParameters {
     // The version of the protocol running.
-    #[serde(rename = "version")]
-    pub(crate) protocol_version: u8,
+    pub(crate) version: u8,
     // The human friendly name of the network.
     #[packable(unpack_error_with = |err| Error::InvalidNetworkName(err.into_item_err()))]
     #[serde(with = "crate::utils::serde::string_prefix")]
-    network_name: StringPrefix<u8>,
+    #[getset(skip)]
+    pub(crate) network_name: StringPrefix<u8>,
     // The HRP prefix used for Bech32 addresses in the network.
-    bech32_hrp: Hrp,
-    // The below max depth parameter of the network.
-    below_max_depth: u8,
+    pub(crate) bech32_hrp: Hrp,
     // The rent structure used by given node/network.
-    rent_structure: RentStructure,
+    pub(crate) rent_structure: RentStructure,
+    // The work score structure used by the node/network.
+    pub(crate) work_score_structure: WorkScoreStructure,
     // TokenSupply defines the current token supply on the network.
     #[serde(with = "crate::utils::serde::string")]
-    token_supply: u64,
+    pub(crate) token_supply: u64,
     // Genesis timestamp at which the slots start to count.
-    #[serde(alias = "genesisUnixTimestamp")]
-    genesis_unix_timestamp: u32,
+    #[serde(with = "crate::utils::serde::string")]
+    pub(crate) genesis_unix_timestamp: u32,
     // Duration of each slot in seconds.
-    #[serde(alias = "slotDurationInSeconds")]
-    slot_duration_in_seconds: u8,
+    pub(crate) slot_duration_in_seconds: u8,
+    pub(crate) slots_per_epoch_exponent: u32,
+    pub(crate) mana_generation_rate: u32,
+    pub(crate) mana_generation_rate_exponent: u32,
+    #[packable(unpack_error_with = |_| Error::InvalidManaDecayFactors)]
```

Comment:
I'm not 100% sure what to do here. The error is infallable, but it's required for the derive

Reply:
Are there no defined bounds in the TIP?

Reply:
it's only defined in the core API, unless I missed it

Reply:
Yeah I don't think there is anything we can do about it atm